### PR TITLE
Disable Vim Mode in Debug Repl

### PR DESCRIPTION
**Pull request by rebornix**
Yay! We love PRs! 🎊

Please include a description of your change & check your PR against this list, thanks:

- [x] Commit message has a short title & issue references
- [x] Each commit does a logical chunk of work.  
- [x] It builds and tests pass (e.g `gulp tslint`)

This PR demonstrates the idea of disabling Vim keybindings in Debug Repl as Debug Repl is now inside the Editor Repl. It doesn't mean we have to merge this in, just want to show team how it works and what's the catch.

### CODE REVIEW

**Comment by jpoon** [comment]
Damn...that is a lot of if/else statements we need for this :(

**Comment by rebornix** [comment]
@jpoon yup, that's for other Code team members to see the dirty workaround with current Code behavior. It's real dirty.

**Comment by johnfn** [comment]
Can you do all of these just once inside keyPressed in modeHandler? Or is there something I'm missing?

**Comment by jpoon** [comment]
> Can you do all of these just once inside keyPressed in modeHandler? 

That's what I was thinking too. But, with the way the code is now, it shows how *bad* of a hack it is and VSCode should really fix it on their end :)

**Comment by rebornix** [comment]
@johnfn As we catch Code's keystrokes like `Backspace`, I don't know how to delegate back this keystroke to Code. If that's possible, then the solution is quite clean somehow. I may need to do some searching or maybe you guys already have ideas.

**Comment by jpoon** [comment]
I believe all keystrokes go through

```
  async handleKeyEvent(key: string): Promise<Boolean> {
```

**Comment by johnfn** [comment]
> it shows how bad of a hack it is and VSCode should really fix it on their end

If @rebornix's solution is really the only way to go about this, IMO it shows more that we architected our own code poorly. :/

**Comment by jpoon** [comment]
Related: https://github.com/VSCodeVim/Vim/issues/666

**Comment by rebornix** [comment]
@jpoon @johnfn now the code looks more reasonable :) We are going to mitigate this issue (like disabling debug panel from editor pane in extension host) this sprint but now this fix can get rid of the issue real quick, even though the additional `inDebugRepl ` check is still dirty.

File: package.json [hunk 1]
```diff
@@ -45,127 +45,127 @@
...
         "key": "right",
         "command": "extension.vim_right",
-        "when": "editorTextFocus && vim.mode != 'Insert Mode'"
+        "when": "editorTextFocus && vim.mode != 'Insert Mode' && !vim.debugInput"
       },
       {
         "key": "up",
         "command": "extension.vim_up",
-        "when": "editorTextFocus && vim.mode != 'Insert Mode'"
+        "when": "editorTextFocus && vim.mode != 'Insert Mode' && !vim.debugInput"
```
Comment (jpoon) [inline package.json]: Is it possible to move this check to a more central location? Maybe inside 'handlekey'?

Comment (rebornix) [inline package.json]: The catch is we can't delegate back modifiers like `ctrl` to Code, that's also why we need to put `vim.mode != 'Insert Mode'` here :(

Comment (jpoon) [inline package.json]: Ah right. Brain fart. 

Change is ugly :( but LGTM.

Comment (jpoon) [inline package.json]: Oh wait. Cause these ctrl modifiers are configured they *should* already go through handlekeys right?

**Pull request merged**